widget/embedded/src/hooks/useSyncNotifications/useSyncNotifications.ts

```diff
@@ -0,0 +1,18 @@
+import { useManager } from '@rango-dev/queue-manager-react';
+
+import { useNotificationStore } from '../../store/notification';
+import { getPendingSwaps } from '../../utils/queue';
+
+export function useSyncNotifications() {
+  const { syncNotifications } = useNotificationStore();
+  const { manager, state } = useManager();
+
+  const swaps = getPendingSwaps(manager);
+
+  const shouldSyncNotifications =
```

Comment:
Currently this code runs in every render and in result of that, 'syncNotifications' runs in every render which seems to be an expensive function. Is this the proper behavior?